### PR TITLE
GOST 28147-89 block cipher implementation (ECB, CBC, CNT and MAC modes included)

### PR DESCRIPTION
**Pull request by zeil**

### CODE REVIEW

File: include/mbedtls/gost89.h [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \brief          Available S-Boxes
+ */
+typedef enum
+{
+    MBEDTLS_GOST89_SBOX_TEST = 0, /**< 1.2.643.2.2.31.0    */
```
Comment (aprelev) [inline include/mbedtls/gost89.h]: Different style of Doxygen comment?
`/**<` (here) vs `/*!<` (on lines 55, 56...)
I'd stick to one.

File: include/mbedtls/gost89.h [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+ */
+void mbedtls_gost89_free( mbedtls_gost89_context *ctx );
+
+/**
+ * \brief          GOST89 key schedule
+ *
+ * \param ctx      GOST89 context to be initialized
+ * \param key      32-byte secret key
+ *
+ * \return         0
```
Comment (aprelev) [inline include/mbedtls/gost89.h]: \return 0 when? Always? On success?

File: include/mbedtls/gost89.h (hunk 1 above)
Comment (zeil) [inline include/mbedtls/gost89.h]: Mbedtls developers use both styles. `/**<` for enums and `/*!<` for struct fields.

File: include/mbedtls/gost89.h (hunk 2 above)
Comment (zeil) [inline include/mbedtls/gost89.h]: Yes, on success.

Comment (aprelev) [inline include/mbedtls/gost89.h]: Write that down in documentation then =)

Comment (zeil) [inline include/mbedtls/gost89.h]: Already done ;)

**Pull request merged**